great_expectations/expectations/metrics/map_metric_provider.py

```diff
@@ -657,6 +710,65 @@ def inner_func(
 
         return wrapper
 
+    elif issubclass(engine, SqlAlchemyExecutionEngine):
+        if partial_fn_type is None:
+            partial_fn_type = MetricPartialFunctionTypes.MAP_CONDITION_SERIES
+        partial_fn_type = MetricPartialFunctionTypes(partial_fn_type)
+        if partial_fn_type not in [MetricPartialFunctionTypes.MAP_CONDITION_SERIES]:
+            raise ValueError(
+                "SqlAlchemyExecutionEngine only supports map_condition_series for multicolumn_condition_partial partial_fn_type"
+            )
+
+        def wrapper(metric_fn: Callable):
+            @metric_partial(
+                engine=engine,
+                partial_fn_type=partial_fn_type,
+                domain_type=domain_type,
+                **kwargs,
+            )
+            @wraps(metric_fn)
+            def inner_func(
+                cls,
+                execution_engine: SqlAlchemyExecutionEngine,
+                metric_domain_kwargs: Dict,
+                metric_value_kwargs: Dict,
+                metrics: Dict[str, Any],
+                runtime_configuration: Dict,
+            ):
+                (
+                    selectable,
+                    compute_domain_kwargs,
+                    accessor_domain_kwargs,
+                ) = execution_engine.get_compute_domain(
+                    domain_kwargs=metric_domain_kwargs, domain_type=domain_type
+                )
+
+                column_list = accessor_domain_kwargs["column_list"]
+                sqlalchemy_engine: sa.engine.Engine = execution_engine.engine
+
+                for column_name in column_list:
+                    if column_name not in metrics["table.columns"]:
+                        raise ge_exceptions.ExecutionEngineError(
+                            message=f'Error: The column "{column_name}" in BatchData does not exist.'
+                        )
+
+                column_clause = [sa.column(cl) for cl in column_list]
+                dialect = execution_engine.dialect_module
+                column_function = metric_fn(
+                    cls,
+                    column_clause,
+                    **metric_value_kwargs,
+                    _dialect=dialect,
+                    _table=selectable,
+                    _sqlalchemy_engine=sqlalchemy_engine,
+                    _metrics=metrics,
+                )
+                return column_function, compute_domain_kwargs, accessor_domain_kwargs
```

Comment:
@NathanFarmer Assuming that I am not missing a section of the code, I am not seeing the following logic:
- Implement the `expected_condition`, followed by
- returning a tuple, where the `sa._not(expected_condition)` is an element.

The reason is that because expectations return unexpected values, we have to compute them.  And the way we compute them is to have the implementation of a metric return a "positive logic" value with the decorator then returning the negation to help with the computation of unexpected values.  Happy to discuss next week.  Thanks!

Reply:
I have made these logic changes and will be including them in the new PR.